Test suite failure on sparc64, loong64

### xrdp version

0.9.24

### Detailed xrdp version, build options

```
See tests/common/test-suite.log
Please report to xrdp-devel@googlegroups.com
============================================================================
make[5]: *** [Makefile:801: test-suite.log] Error 1
make[5]: Leaving directory '/<<PKGBUILDDIR>>/tests/common'
make[4]: *** [Makefile:909: check-TESTS] Error 2
make[4]: Leaving directory '/<<PKGBUILDDIR>>/tests/common'
make[3]: *** [Makefile:982: check-am] Error 2
make[3]: Leaving directory '/<<PKGBUILDDIR>>/tests/common'
make[2]: *** [Makefile:397: check-recursive] Error 1
make[2]: Leaving directory '/<<PKGBUILDDIR>>/tests'
make[1]: *** [Makefile:483: check-recursive] Error 1
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
dh_auto_test: error: make -j4 check "TESTSUITEFLAGS=-j4 --verbose" VERBOSE=1 returned exit code 2
make: *** [debian/rules:61: binary-arch] Error 25
```

### Operating system & version

Debian GNU/Linux sid

### Installation method

git clone & make install

### Which backend do you use?

_No response_

### What desktop environment do you use?

amiwm

### Environment xrdp running on

physical machine

### What's your client?

Microsoft Remote Desktop on macOS and rdesktop and xfreerdp

### Area(s) with issue?

Compile error

### Steps to reproduce

Please see https://buildd.debian.org/status/package.php?p=xrdp&suite=experimental

not yet available .dsc link, but basically dget -x url.to.the.dsc
available now from: 

```
dget -x http://sid.ethz.ch/debian/xrdp/2023/xrdp_0.9.24-1.dsc
cd xrdp*/
debuild
```

### ✔️ Expected Behavior

it builds

### ❌ Actual Behavior

FTBFS

### Anything else?

_No response_

--- a/tests/common/test_ssl_calls.c
+++ b/tests/common/test_ssl_calls.c
@@ -9,6 +9,11 @@
 
 #include "test_common.h"
 
+/* Time to allow RSA-based test suites to run on older, slower platforms
+ *
+ * These platforms are most often seen on build farms (e.g. Debian CI) */
+#define RSA_BASED_TEST_SUITE_TIMEOUT 60
+
 START_TEST(test_rc4_enc_ok)
 {
     const char *key = "16_byte_key-----";
@@ -369,23 +374,39 @@ Suite *
 make_suite_test_ssl_calls(void)
 {
     Suite *s;
-    TCase *tc_ssl_calls;
+    TCase *tc;
 
     s = suite_create("SSL-Calls");
 
-    tc_ssl_calls = tcase_create("ssl_calls");
-    suite_add_tcase(s, tc_ssl_calls);
-    tcase_add_test(tc_ssl_calls, test_rc4_enc_ok);
-    tcase_add_test(tc_ssl_calls, test_rc4_enc_tv0_ok);
-    tcase_add_test(tc_ssl_calls, test_rc4_enc_tv1_ok);
-    tcase_add_test(tc_ssl_calls, test_rc4_enc_tv2_ok);
-    tcase_add_test(tc_ssl_calls, test_rc4_enc_tv3_ok);
-    tcase_add_test(tc_ssl_calls, test_rc4_enc_tv4_ok);
-    tcase_add_test(tc_ssl_calls, test_sha1_hash_ok);
-    tcase_add_test(tc_ssl_calls, test_md5_hash_ok);
-    tcase_add_test(tc_ssl_calls, test_des3_enc_ok);
-    tcase_add_test(tc_ssl_calls, test_hmac_sha1_dgst_ok);
-    tcase_add_test(tc_ssl_calls, test_gen_key_xrdp1);
+    tc = tcase_create("ssl_calls_rc4");
+    suite_add_tcase(s, tc);
+    tcase_add_test(tc, test_rc4_enc_ok);
+    tcase_add_test(tc, test_rc4_enc_tv0_ok);
+    tcase_add_test(tc, test_rc4_enc_tv1_ok);
+    tcase_add_test(tc, test_rc4_enc_tv2_ok);
+    tcase_add_test(tc, test_rc4_enc_tv3_ok);
+    tcase_add_test(tc, test_rc4_enc_tv4_ok);
+
+    tc = tcase_create("ssl_calls_sha1");
+    suite_add_tcase(s, tc);
+    tcase_add_test(tc, test_sha1_hash_ok);
+
+    tc = tcase_create("ssl_calls_md5");
+    suite_add_tcase(s, tc);
+    tcase_add_test(tc, test_md5_hash_ok);
+
+    tc = tcase_create("ssl_calls_des3");
+    suite_add_tcase(s, tc);
+    tcase_add_test(tc, test_des3_enc_ok);
+
+    tc = tcase_create("ssl_calls_hmac_sha1");
+    suite_add_tcase(s, tc);
+    tcase_add_test(tc, test_hmac_sha1_dgst_ok);
+
+    tc = tcase_create("ssl_calls_rsa_key");
+    suite_add_tcase(s, tc);
+    tcase_set_timeout(tc, RSA_BASED_TEST_SUITE_TIMEOUT);
+    tcase_add_test(tc, test_gen_key_xrdp1);
 
     return s;
 }

--- a/tests/xrdp/test_bitmap_load.c
+++ b/tests/xrdp/test_bitmap_load.c
@@ -45,6 +45,9 @@ static int WHITE = COLOR24RGB(255, 255, 255);
  * be considered close enough to each other */
 #define MAX_SIMILAR_COLOR_DISTANCE 3
 
+/* Time to allow some large bitmap test suites to run on slower platforms */
+#define LARGE_BM_TEST_SUITE_TIMEOUT 10
+
 void setup(void)
 {
 }
@@ -301,37 +304,48 @@ Suite *
 make_suite_test_bitmap_load(void)
 {
     Suite *s;
-    TCase *tc_bitmap_load;
-#ifdef USE_IMLIB2
-    TCase *tc_other_load;
-#endif
+    TCase *tc;
 
     s = suite_create("BitmapLoad");
 
-    tc_bitmap_load = tcase_create("xrdp_bitmap_load");
-    tcase_add_checked_fixture(tc_bitmap_load, setup, teardown);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__with_invalid_image__fail);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__4_bit__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__8_bit__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__24_bit__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__max_width_zoom__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__max_height_zoom__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__min_zoom__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__max_width_scale__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__max_height_scale__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__min_scale__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__not_4_pixels_wide_4_bit__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__not_4_pixels_wide_8_bit__ok);
-    tcase_add_test(tc_bitmap_load, test_bitmap_load__not_4_pixels_wide_24_bit__ok);
+    tc = tcase_create("xrdp_bitmap_load_basic");
+    suite_add_tcase(s, tc);
+    tcase_add_checked_fixture(tc, setup, teardown);
+    tcase_add_test(tc, test_bitmap_load__with_invalid_image__fail);
+    tcase_add_test(tc, test_bitmap_load__4_bit__ok);
+    tcase_add_test(tc, test_bitmap_load__8_bit__ok);
+    tcase_add_test(tc, test_bitmap_load__24_bit__ok);
 
-    suite_add_tcase(s, tc_bitmap_load);
+    tc = tcase_create("xrdp_bitmap_load_zoom");
+    suite_add_tcase(s, tc);
+    tcase_add_checked_fixture(tc, setup, teardown);
+    tcase_set_timeout(tc, LARGE_BM_TEST_SUITE_TIMEOUT);
+    tcase_add_test(tc, test_bitmap_load__max_width_zoom__ok);
+    tcase_add_test(tc, test_bitmap_load__max_height_zoom__ok);
+    tcase_add_test(tc, test_bitmap_load__min_zoom__ok);
+
+    tc = tcase_create("xrdp_bitmap_load_scale");
+    suite_add_tcase(s, tc);
+    tcase_add_checked_fixture(tc, setup, teardown);
+    tcase_set_timeout(tc, LARGE_BM_TEST_SUITE_TIMEOUT);
+    tcase_add_test(tc, test_bitmap_load__max_width_scale__ok);
+    tcase_add_test(tc, test_bitmap_load__max_height_scale__ok);
+    tcase_add_test(tc, test_bitmap_load__min_scale__ok);
+
+    tc = tcase_create("xrdp_bitmap_load_not_mod_4");
+    suite_add_tcase(s, tc);
+    tcase_add_checked_fixture(tc, setup, teardown);
+    tcase_add_test(tc, test_bitmap_load__not_4_pixels_wide_4_bit__ok);
+    tcase_add_test(tc, test_bitmap_load__not_4_pixels_wide_8_bit__ok);
+    tcase_add_test(tc, test_bitmap_load__not_4_pixels_wide_24_bit__ok);
+
 
 #ifdef USE_IMLIB2
-    tc_other_load = tcase_create("xrdp_other_load");
-    tcase_add_checked_fixture(tc_other_load, setup, teardown);
-    tcase_add_test(tc_other_load, test_png_load__blend_ok);
-    tcase_add_test(tc_other_load, test_jpg_load__ok);
-    suite_add_tcase(s, tc_other_load);
+    tc = tcase_create("xrdp_other_load");
+    suite_add_tcase(s, tc);
+    tcase_add_checked_fixture(tc, setup, teardown);
+    tcase_add_test(tc, test_png_load__blend_ok);
+    tcase_add_test(tc, test_jpg_load__ok);
 #endif
 
     return s;

